src/d3a/models/market/two_sided.py

```diff
@@ -264,9 +277,70 @@ def match_recommendation(self, recommended_list):
                     recommended_list, index+1, trade, bid_trade
                 )
 
+    @staticmethod
+    def validate_requirements_satisfied(obj1: Union[Offer, Bid], obj2: Union[Offer, Bid]) -> None:
+        """Validate if both trade parties satisfy each other's requirements.
+
+        :raises:
+            InvalidBidOfferPair: Bid offer pair failed the validation
+        """
+        def _validate_requirements_satisfied(_obj1: Union[Offer, Bid], _obj2: Union[Offer, Bid]):
+            if _obj1.requirements:
```

Comment:
In order to reduce the number of indentions here I would rather return early here
```suggestion
            if not _obj1.requirements:
                return
```